src/components/containers/ConnectedMellowLpDepositForm/ConnectedMellowLpDepositForm.tsx

```diff
@@ -0,0 +1,221 @@
+    APPROVE_REQUIRED = "APPROVE_REQUIRED",
+    APPROVING = "APPROVING",
+    APPROVE_FAILED = "APPROVE_FAILED",
+    APPROVED = "APPROVED",
+    DEPOSITING = "DEPOSITING",
+    DEPOSIT_FAILED = "DEPOSIT_FAILED",
+    DEPOSIT_DONE = "DEPOSIT_DONE",
+};
+
+type TextProps = {
+    bold?: boolean;
+    children?: ReactNode;
+    green?: boolean;
+    red?: boolean;
+};
+const Text = ({ bold, children, green, red }: TextProps) => (
+    <Box component='span' sx={{
+        color: green ? colors.vzCustomGreen1 : red ? colors.vzCustomRed1 : undefined,
+        fontWeight: bold ? 'bold' : undefined,
+        textTransform: 'none'
+    }}>
+        {children}
+    </Box>
+);
+
+const ConnectedMellowLpDepositForm: React.FunctionComponent<ConnectedMellowLpDepositFormProps> = ({ vault, onReset }) => {
+    const navigate = useNavigate();
+
+    const [selectedDeposit, setSelectedDeposit] = useState<number>(0);
+
+    const [depositState, setDepositState] = useState<DepositStates>(DepositStates.INITIALISING);
+    const [error, setError] = useState<ReactNode>();
+
+    const sufficientFunds = (vault.userWalletBalance ?? 0) >= selectedDeposit;
+
+    const deposit = () => {
+        if (selectedDeposit > 0) {
+            setDepositState(DepositStates.DEPOSITING);
+            void vault.deposit(selectedDeposit).then(() => {
+                setDepositState(DepositStates.DEPOSIT_DONE);
+            }, (err: Error) => {
+                setError((<Text red>{`Deposit failed. ${err.message ?? ""}`}</Text>));
+                setDepositState(DepositStates.DEPOSIT_FAILED);
+            })
+        }
+        else {
+            setError("Please input amount");
+            setDepositState(DepositStates.DEPOSIT_FAILED);
+        }
+    };
+
+    const approve = () => {
+        setDepositState(DepositStates.APPROVING);
+        void vault.approveToken().then(() => {
+            setDepositState(DepositStates.APPROVED);
+        }, (err: Error) => {
+            setError((<Text red>{`Approval failed. ${err.message ?? ""}`}</Text>));
+            setDepositState(DepositStates.APPROVE_FAILED);
+        })
+    };
+
+    useEffect(() => {
+        void vault.isTokenApproved().then((resp) => {
+            if (resp) {
+                setDepositState(DepositStates.APPROVED);
+            }
+            else {
+                setDepositState(DepositStates.APPROVE_REQUIRED);
+            }
+        }, (err: Error) => {
+            setError((<Text red> {`Error occured while retrieving information. ${err.message ?? ""}`}</Text>));
+            setDepositState(DepositStates.PROVIDER_ERROR);
+        });
+    }, [vault, selectedDeposit]);
+
+    let submissionState: {
+        submitText: string,
+        hintText: ReactNode,
+        action: () => void,
+        disabled: boolean,
+    } = {
+        submitText: "Initialising...",
+        action: () => { },
+        disabled: true,
+        hintText: "Initialising, please wait",
+    }
+
+    switch (depositState) {
```

Comment:
This looks like a mapper. I would consider moving it to another file living in same directory as the component. Perhaps ./mappers.ts

Reply:
Resolved. https://github.com/Voltz-Protocol/voltz-ui/pull/295/commits/a44c3c48e2a2660e1fc14f6eeeda646152d2edea